totoify_grafeas/totoifylib.py

```diff
@@ -0,0 +1,277 @@
+                                          in_toto_link.signed.command,
+                                          in_toto_link.signed.byproducts,
+                                          in_toto_link.signed.environment)
+      for signature in in_toto_link.signatures:
+        self.intoto["signatures"].append({"keyid": signature["keyid"], "signature": signature["sig"]})
+
+    self.note_name = note_name
+
+    self.resource["uri"] = resource_uri
+
+  def to_json(self, file_path=None):
+    """Returns the JSON string representation."""
+    if file_path:
+      with open(file_path, "w") as fp:
+        json.dump({
+            "resource": self.resource,
+            "noteName": self.note_name,
+            "kind": self.kind,
+            "intoto": {
+              "signatures": self.intoto["signatures"],
+              "signed": self.intoto["signed"].to_dict()
+            }
+          }, fp, indent=4)
+    else:
+      return json.dumps({
+          "resource": self.resource,
+          "noteName": self.note_name,
+          "kind": self.kind,
+          "intoto": {
+            "signatures": self.intoto["signatures"],
+            "signed": self.intoto["signed"].to_dict()
+          }
+        }
+      )
+
+  @staticmethod
+  def load(path):
+    """Given a path, loads the occurrence.json into GrafeasInTotoOccurrence class."""
+    with open(path, "r") as occ_f:
+      occ_json = json.load(occ_f)
+
+    grafeas_occurrence = GrafeasInTotoOccurrence()
+    grafeas_occurrence.intoto['signed'] = GrafeasLink(occ_json['intoto']['signed']['materials'],
+                                                      occ_json['intoto']['signed']['products'],
+                                                      occ_json['intoto']['signed']['command'],
+                                                      occ_json['intoto']['signed']['byproducts'],
+                                                      occ_json['intoto']['signed']['environment'])
+    grafeas_occurrence.intoto['signatures'] = occ_json['intoto']['signatures']
+    grafeas_occurrence.resource['uri'] = occ_json['resource']['uri']
+    grafeas_occurrence.noteName = occ_json['noteName']
+    return grafeas_occurrence
+
+
+
+def create_grafeas_occurrence_from_in_toto_link(in_toto_link, step_name, resource_uri):
+  """Creates a GrafeasInTotoOccurrence class from an in-toto link Metablock class.
+
+  Arguments:
+    in_toto_link: A link of type Metablock (and we know that "signed" corresponds
+        to an object of type in_toto.models.link.Link)
+    step_name: A string of the step name
+    resource_uri: A string of the resource URI
+
+  Raises:
+    N/A
+
+  Side Effects:
+    N/A
+
+  Returns:
+    A GrafeasInTotoOccurrence object.
+  """
+  # TODO: set step_name from link if None
+  return GrafeasInTotoOccurrence(in_toto_link, step_name, resource_uri)
+
+
+
+def create_in_toto_link_from_grafeas_occurrence(grafeas_occurrence, step_name):
```

Comment:
I think the same could be done here (but instead we have a instance method):

```
grafeasOccurrence = GrafeasInTotoOcurrence(...)
grafeasOccurrence.to_link()
```
I also think it'd be a little simpler to follow when doing something like this:

```
my_link = ...
grafeas_occurrence = GrafeasInTotoOcurrence(my_link)
assertTrue(my_link == grafeas_occurrence.to_link())
```